packages/adaptive_scaffold/lib/src/adaptive_layout.dart

```diff
@@ -0,0 +1,513 @@
+            beginWidth = halfWidth - leftMargin;
+          } else {
+            beginWidth = remainingWidth * bodyRatio!;
+          }
+          currentBodySize = layoutChild(
+              _bodyID,
+              BoxConstraints.tight(Size(
+                  animatedSize(beginWidth, remainingWidth), remainingHeight)));
+        } else {
+          double beginHeight;
+          if (bodyRatio == null) {
+            beginHeight = halfHeight - topMargin;
+          } else {
+            beginHeight = remainingHeight * bodyRatio!;
+          }
+          currentBodySize = layoutChild(
+              _bodyID,
+              BoxConstraints.tight(Size(
+                  remainingWidth, animatedSize(beginHeight, remainingHeight))));
+        }
+        layoutChild(_secondaryBodyID, BoxConstraints.loose(size));
+      } else {
+        if (horizontalBody) {
+          // If body and secondaryBody laid out horizontally
+          if (textDirection) {
+            // If textDirection is LTR
+
+            double finalBodySize;
+            double finalSBodySize;
+            if (hinge != null) {
+              finalBodySize = hinge!.left - leftMargin;
+              finalSBodySize =
+                  size.width - (hinge!.left + hingeWidth) - rightMargin;
+            } else if (bodyRatio != null) {
+              finalBodySize = remainingWidth * bodyRatio!;
+              finalSBodySize = remainingWidth * (1 - bodyRatio!);
+            } else {
+              finalBodySize = halfWidth - leftMargin;
+              finalSBodySize = halfWidth - rightMargin;
+            }
+
+            currentBodySize = layoutChild(
+                _bodyID,
+                BoxConstraints.tight(Size(
+                    animatedSize(remainingWidth, finalBodySize),
+                    remainingHeight)));
+            layoutChild(_secondaryBodyID,
+                BoxConstraints.tight(Size(finalSBodySize, remainingHeight)));
+          } else {
+            // If textDirection is RTL
+
+            double finalBodySize;
+            double finalSBodySize;
+            if (hinge != null) {
+              finalBodySize =
+                  size.width - (hinge!.left + hingeWidth) - rightMargin;
+              finalSBodySize = hinge!.left - leftMargin;
+            } else if (bodyRatio != null) {
+              finalBodySize = remainingWidth * bodyRatio!;
+              finalSBodySize = remainingWidth * (1 - bodyRatio!);
+            } else {
+              finalBodySize = halfWidth - rightMargin;
+              finalSBodySize = halfWidth - leftMargin;
+            }
+            currentSBodySize = layoutChild(
+                _secondaryBodyID,
+                BoxConstraints.tight(
+                    Size(animatedSize(0, finalSBodySize), remainingHeight)));
+            layoutChild(_bodyID,
+                BoxConstraints.tight(Size(finalBodySize, remainingHeight)));
+          }
+        } else {
+          // If body and secondaryBody laid out vertically
```

Comment:
```suggestion
          // Take this path if the body and secondaryBody are laid out vertically.
```

Try for complete sentences for all comments.

https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#use-correct-grammar